### PR TITLE
Add --auth to allow GH personal tokens.

### LINKED ISSUE #68: Feat: Basic Authentication for Github API
**Description**

At this time, we're having issues hitting rate limits - especially when testing. GitHub only allows 10 requests/minute. [Basic Authentication would allow up to 30/minute](https://docs.github.com/en/rest/reference/search#rate-limit).

**Why is it necessary? (or how would it make it better)**

This is necessary to expand our ability to perform automated tests without those tests failing due to Rate Limit issues.

**how do you think should it be implemented? (If possible)**

I think we should use a username/api key in the request header for [Basic Authentication](https://docs.github.com/en/rest/overview/other-authentication-methods#basic-authentication). We could then just place the appropriate credentials in an environment variable and [use click to access it](https://click.palletsprojects.com/en/7.x/arguments/?highlight=environment%20variables#environment-variables). We could also cache the username/api key, which I presume we would want to do securely... Not sure how to do that without external modules or if our automated testing would support that.

I would love to work on this! I don't know much about the automated PR testing, however. I would need to know if we can run it with the env variable and which account credentials we would use for testing.

### PR DESCRIPTION
**Pull request by jSadoski**
**Context**

Resolves #68 

**Description**

Allows personal token using `starcli --auth *username:password*`. Validates option & provides a message if submitted in the wrong format. The `status_action` for `http_code_handling['401']` was changed to a new `status_action`: unauthorized. DEBUG output was also added for the option 👍

P.S. I ran the tests with this and I was able to complete them all! It seems the auth + the new retry technique is working.

### CODE REVIEW

File: starcli/__main__.py [hunk 1]
```diff
@@ -89,6 +90,12 @@
     help="Search for trending repositories by username",
 )
 @click.option("--debug", is_flag=True, default=False, help="Turn on debugging mode")
+@click.option(
+    "--auth", 
+    type=str, 
+    default="", 
+    help="GitHub personal access token in the format 'username:token'. This can help with rate limits. See: https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token"
```
Comment (hedythedev) [inline starcli/__main__.py:97]: Thanks! The sentence ”This can help with rate limits...” and after that can be removed

Comment (hedythedev) [reply starcli/__main__.py:97]: The docs link can be added in the readme when needed

**Comment by hedythedev** [comment]
The code good, there are some lines which has a very wide width, could you please run `black .` or is it `pipenv run black .`? I’m not familiar with pipenv

**Comment by jSadoski** [comment]
I switched back to virtualenv so my env is more in line with the other development here.

**Comment by hedythedev** [comment]
ok then it’s `python -m black .`

**Comment by hedythedev** [comment]
Also, did you have a go with storing the credentials in a file? Maybe authenticating using token will be safer because this is using username and password.

**Comment by jSadoski** [comment]
I did not go with storing the credentials in a file, yet. That's a feature I'd like to do soon, but I submitted this because it allows the full testing suite to run so we can get cli() tests up ASAP.

I agree tokens are better. Right now, GitHub allows authentication using 'username:token' OR 'username:password' in the request header. [GitHub is deprecating 'username:password' on November 13 of this year](https://developer.github.com/changes/2020-02-14-deprecating-password-auth/). This code was intended to be used with a  personal access token, but since GitHub allows the password in the header, the request is being authenticated when the password is used, too. I think we could sort of check to make sure the token is being used by checking the length of the the string passed, unless of course their password were the same length as a token.

Otherwise, instead of basic authentication, we could go straight to implementing [OAuth2 Device Flow](https://docs.github.com/en/developers/apps/authorizing-oauth-apps#device-flow) instead. This would save users the step of generating a personal token and would be more secure.

**Comment by hedythedev** [comment]
@allcontributors please add @jSadoski for code

**Pull request merged**